mlir/lib/Analysis/Presburger/IntegerRelation.cpp

```diff
@@ -1088,10 +1084,10 @@ unsigned IntegerRelation::gaussianEliminateVars(unsigned posStart,
   unsigned pivotCol = 0;
   for (pivotCol = posStart; pivotCol < posLimit; ++pivotCol) {
     // Find a row which has a non-zero coefficient in column 'j'.
-    unsigned pivotRow;
-    if (!findConstraintWithNonZeroAt(pivotCol, /*isEq=*/true, &pivotRow)) {
+    std::optional<unsigned> pivotRow;
+    if (!(pivotRow = findConstraintWithNonZeroAt(pivotCol, /*isEq=*/true))) {
       // No pivot row in equalities with non-zero at 'pivotCol'.
-      if (!findConstraintWithNonZeroAt(pivotCol, /*isEq=*/false, &pivotRow)) {
+      if (!(pivotRow = findConstraintWithNonZeroAt(pivotCol, /*isEq=*/false))) {
```

Comment:
Worth noting that this value of pivotRow from the second call isn't used here, so the assignment isn't needed. Here and above it's cleaner IMO to assign the optional directly and check !optional

Reply:
Sure, done.